core/gather/navigation-runner.js

```diff
@@ -227,6 +227,11 @@ async function _navigation(navigationContext) {
   };
 
   const setupResult = await _setupNavigation(navigationContext);
+
+  // TODO: Do this as part of the DevtoolsLog gatherer.
```

Comment:
it feels weird to be refactoring this way but also having a TODO to refactor it a different way. Does this TODO mean to look into how to resolve the scheduling differences between the gatherers so it can be part of the DevtoolsLog gatherer? (one solution to those issues is to not do it at the gatherer level and do it at the runner level instead, like this :)

Reply:
Yeah this todo is definitely missing a lot of detail...

I think the best path forward is to remove this TODO and file an issue that better summarizes these concerns. I could see us keeping this solution long term but there are still things we need to consider from #14717